langfuse-core/src/index.ts

```diff
@@ -564,46 +569,54 @@ abstract class LangfuseCoreStateless {
    *** QUEUEING AND FLUSHING
    ***/
   protected enqueue(type: LangfuseObject, body: EventBody): void {
-    try {
-      if (!this.enabled) {
-        return;
-      }
+    if (!this.enabled) {
+      return;
+    }
 
-      this.maskEventBodyInPlace(body);
-      const finalEventBody = this.truncateEventBody(body, MAX_EVENT_SIZE);
+    const promise = this.processEnqueueEvent(type, body);
+    const promiseId = generateUUID();
+    this.pendingEventProcessingPromises[promiseId] = promise;
 
-      try {
-        JSON.stringify(finalEventBody);
-      } catch (e) {
-        console.error(`[Langfuse SDK] Event Body for ${type} is not JSON-serializable: ${e}`);
-        this._events.emit("error", `Event Body for ${type} is not JSON-serializable: ${e}`);
+    promise
+      .catch((e) => {
+        this._events.emit("error", e);
+      })
+      .finally(() => {
+        delete this.pendingEventProcessingPromises[promiseId];
+      });
+  }
 
-        return;
-      }
+  protected async processEnqueueEvent(type: LangfuseObject, body: EventBody): Promise<void> {
+    this.maskEventBodyInPlace(body);
+    await this.processMediaInEvent(type, body);
```

Comment:
Here, we create the promise to upload when we add an event to the ingestion queue. This is a bit unexpected. I would have expected this to happen in the logic in the flush function.